invokeai/app/invocations/noise.py

```diff
@@ -69,13 +66,13 @@ class NoiseOutput(BaseInvocationOutput):
     width: int = OutputField(description=FieldDescriptions.width)
     height: int = OutputField(description=FieldDescriptions.height)
 
-
-def build_noise_output(latents_name: str, latents: torch.Tensor, seed: int):
-    return NoiseOutput(
-        noise=LatentsField(latents_name=latents_name, seed=seed),
-        width=latents.size()[3] * 8,
-        height=latents.size()[2] * 8,
-    )
+    @classmethod
+    def build(cls, latents_name: str, latents: torch.Tensor, seed: int) -> "NoiseOutput":
+        return cls(
+            noise=LatentsField(latents_name=latents_name, seed=seed),
+            width=latents.size()[3] * 8,
```

Comment:
May make sense to make a downsampling factor optional param here in case a node has latents made with a different one ```downsampling_factor: int = 8```

Reply:
Yeah, good call.

Reply:
On second thought, I think adding an additional arg that is never used. I'd prefer to wait for a need for this to add that functionality.

Reply:
I was also made uncomfortable by this at one point and created this constant to ease my discomfort: https://github.com/invoke-ai/InvokeAI/blob/f64fc2c8b76985c68e6ecba7872020e53d51a6fa/invokeai/app/invocations/latent.py#L82-L86

Feel free to use it if you want. But also fine if you leave it since this is not the focus of this PR.

Reply:
I've added the constant, thanks. If we need to make it dynamic in the future we can always update the API - wanna keep the method signatures as minimal as possible.

Reply:
Well, that creates a circular reference, reverted. I've made a ticket to remind me to follow up.